src/alpaca_parser.yrl

```diff
@@ -449,11 +452,20 @@ tuple -> '(' tuple_list ')' :
 
 infix -> term op term : make_infix('$2', '$1', '$3').
 
+literal_fun -> fn terms '->' simple_expr:
```

Comment:
Doesn't this mean you could have a lambda in the argument position of another lambda? Perhaps "term" needs to be split into terms that are safe to pattern match on and terms that are not.

To clarify:
````
let add1 = fn (fn x -> x + 1) -> 2
````
Parses, but later crashes in alpaca_codegen.